test/gui/shared/steps/steps.py

```diff
@@ -666,21 +500,13 @@ def step(context, permissions, resource):
 )
 def step(context, permissions, resource, password):
     resource = sanitizePath(substituteInLineCodes(context, resource))
-    openPublicLinkDialog(context, resource)
-    clickButton(waitForObject(names.oCC_ShareLinkWidget_checkBox_password_QCheckBox))
-    mouseClick(
-        waitForObject(names.oCC_ShareLinkWidget_lineEdit_password_QLineEdit),
-        0,
-        0,
-        Qt.NoModifier,
-        Qt.LeftButton,
-    )
-    type(waitForObject(names.oCC_ShareLinkWidget_lineEdit_password_QLineEdit), password)
-    clickButton(waitForObject(names.oCC_ShareLinkWidget_createShareButton_QPushButton))
-    waitFor(
-        lambda: (
-            findObject(names.linkShares_0_0_QModelIndex).displayText == "Public link"
-        )
+    openSharingDialog(context, resource)
+    publicLinkDialog = PublicLinkDialog()
+    publicLinkDialog.openPublicLinkDialog()
+    radioObjectName = getRadioObjectForPermssion(permissions)
+
+    publicLinkDialog.createPublicLink(
+        context, resource, password, radioObjectName, permissions
     )
```

Comment:
instead of radio object name pass permissions to the functions and get the name inside the page objects.

Reply:
Done